packages/blocks/src/__internal__/utils/components.ts

```diff
@@ -11,38 +11,48 @@ import type { EmbedBlockModel } from '../../embed-block';
 import type { ShapeBlockModel } from '../../shape-block';
 
 // TODO support dynamic block types
-export function BlockElement(
-  model: BaseBlockModel,
-  host: BlockHost,
-  edgeless = false
-) {
+export function BlockElement({
+  model,
+  host,
+  edgeless = false,
+  readonly = false,
+}: {
+  model: BaseBlockModel;
+  host: BlockHost;
+  edgeless?: boolean;
+  readonly?: boolean;
+}) {
   switch (model.flavour) {
     case 'affine:paragraph':
       return html`
         <paragraph-block
           .model=${model as ParagraphBlockModel}
           .host=${host}
+          .readonly=${readonly}
         ></paragraph-block>
       `;
     case 'affine:list':
       return html`
         <list-block
           .model=${model as ListBlockModel}
           .host=${host}
+          .readonly=${readonly}
```

Comment:
I'm not sure if we should add arbitrary new props in this way. Can we `host.readonly` instead?